packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1136,6 +1136,40 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
   @mustCallSuper
   void deactivate() { }
 
+  /// Called when this object is reinserted into the tree after having been
+  /// removed via [deactivate].
+  ///
+  /// In most cases, after a [State] object has been deactivated, it is _not_
+  /// reinserted into the tree, and its [dispose] method will be called to
+  /// signal that it is ready to be garbage collected.
+  ///
+  /// In some cases, however, after a [State] object has been deactivated, the
+  /// framework will reinsert it into another part of the tree (e.g., if the
+  /// subtree containing this [State] object is grafted from one location in
+  /// the tree to another). If that happens, the framework will call [activate]
```

Comment:
nit: maybe link this somehow to globalkeys as an example of what can trigger this in practice?